packages/app/src/components/nominationInfo/EditCard.js

```diff
@@ -0,0 +1,147 @@
+import DatePicker from 'react-datepicker';
+import styles from './newstyles.module.css';
+import { Controller } from 'react-hook-form';
+import { Link } from 'react-router-dom';
+import './calendar.css';
+
+const EditCard = props => {
+  // state for React Datepicker
+  const [admissionDate, setAdmissionDate] = useState(
+    new Date(props.formData['Admission Date'])
+  );
+
+  // state for React Datepicker
+  const [dischargeDate, setDischargeDate] = useState(
+    new Date(props.formData['Discharge Date'])
+  );
+
+  // keys for our form data object (passed down from ApplicationForm.js)
+  let keys = Object.keys(props.formData);
+  return (
+    <div className={styles.card}>
+      <div className={[styles.gridContainer, styles.content].join(' ')}>
+        {keys.map(label => {
+          switch (true) {
+            // Render our titles in form
+            case props.titleLabels.includes(label):
+              return (
+                <div className={styles.header}>
+                  <div key={label} className={styles.title}>
+                    {label}
+                  </div>
+                </div>
+              );
+            // Render dates
+            case props.editableDates.includes(label):
+              return (
+                <div>
+                  <label className={styles.label}>{label}</label>
+                  {/* Controller Component needed for React Hook Form and React Datepicker Integration*/}
+                  <Controller
+                    name={label}
+                    control={props.control}
+                    defaultValue={
+                      // from useState hooks above
+                      admissionDate && dischargeDate
+                        ? label === 'Admission Date'
+                          ? new Date(admissionDate)
+                          : new Date(dischargeDate)
+                        : null
+                    }
+                    render={({ field: { onChange, value } }) => (
+                      <DatePicker
+                        selected={value}
+                        onChange={onChange}
+                        dateFormat='MM/dd/yyyy'
+                      />
+                    )}
+                  />
+                  <p className={styles.yupError}>
+                    {props.errors[label]?.message}
+                  </p>
+                </div>
+              );
+              {
+                /* Renders the rest of the form */
+              }
+            case props.editablePlainText.includes(label):
+              return (
+                <div>
+                  <label className={styles.label}>{label}</label>
+                  <input
+                    name={label}
+                    type='text'
+                    defaultValue={props.formData[label]}
+                    {...props.register(label)}
+                    className={
+                      props.errors[label]?.message ? styles.inputError : ''
```

Comment:
these errors logics are to check if the errors object exists (it does not exist if yup validation is successful) i don't know how it could be removed from here.

Reply:
I think it's fine to leave this if there's no automatic error styling that occurs.

Reply:
hell yeah

Reply:
lol